### PR TITLE
Album cleanup

### PR DESCRIPTION
**Pull request by zas**
# Summary

* This is a…
  * [ ] Bug fix
  * [ ] Feature addition
  * [x] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): PICARD-XXX

# Solution

# Action

### CODE REVIEW

File: picard/album.py [hunk 1]
```diff
@@ -645,12 +645,7 @@ def is_complete(self):
             return True
 
     def is_modified(self):
-        if self.tracks:
-            for track in self.tracks:
-                for file in track.files:
-                    if not file.is_saved():
-                        return True
-        return False
+        return self.get_num_unsaved_files() > 0
```
Comment (phw) [inline picard/album.py:648]: Not sure about this. It saves us a few lines of code, but for long list of tracks returning on first modified is a serious performance improvement. What about something like this (untested)?

```python
    def is_modified(self):
        for file in self._iter_unsaved_files():
            return True
        return False

    def get_num_unsaved_files(self):
        count = 0
        for file in self._iter_unsaved_files():
            count += 1
        return count

    def _iter_unsaved_files(self):
        for track in self.tracks:
            for file in track.files:
                if not file.is_saved():
                    yield file
```

Comment (phw) [reply picard/album.py:648]: Or even for _iter_unsaved_files:

```python
    def _iter_unsaved_files(self):
        for file in self.iterfiles(save=True):
            if not file.is_saved():
                yield file

**Review by phw — CHANGES_REQUESTED** [review]

File: picard/album.py (hunk 1 above)
Comment (zas) [reply picard/album.py]: Yes, you're right, that's much better. Done in f3cb6735458179558ae82910efb33d1424b0c562

**Comment by zas** [comment]
I re-organized commits to include changes suggested by @phw

**Review by phw — APPROVED** [review]

**Pull request merged**